File: Computer Science - OSN Informatika - Python - C++/Private Axel/Dynamic Programming Intro Very Basic/recursive.cpp
#include <iostream>
using namespace std;
int f(int n){
    if(n == 1 || n == 2) return 1; // 1.Base case f(1) = 1, f(2) = 1
    return f(n - 1) + f(n - 2); // 2. Rekurens f(n) = f(n - 1) + f(n - 2)
}

int main(){
    int N;
    cin>>N;
    cout<<f(N);
    return 0;
}

File: Computer Science - OSN Informatika - Python - C++/Privat Wayan/recursive.py
# 1) Nama fungsinya
# 2) Base case
# 3) Rekurens
# faktorial basecase 1! = 1, 0! = 1
def faktorial(n):
    if(n == 1):
        return 1
    else:
        return n * faktorial(n - 1)

T = int(input())
for _ in range(T):
    N = int(input())
    print(faktorial(N))

